pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -326,6 +327,12 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(
 			allVolumesAdded = false
 			continue
 		}
+		uniqueVolumeName, exist := getUniqueVolumeName(uniquePodName, podVolume.Name, mountedVolumesForPod)
```

Comment:
getUniqueVolumeName is called ehre, and then the following AddPodToVolume also calls the same function getUniqueVolumeName. Better to remove the redundancy as I mentioned before.

Reply:
There are two getUniqueVolumeName func's with different signatures - one in the populator and one in dsw.

They also have different logic - the one in populator looks up in map, the one in dsw calls util.GetUniqueVolumeNameFromSpec

Reply:
ok, I was confused by those two functions with the same name. So getUniqueVolumeName here is checking whether volume has been mounted and added to actual state. I don't think you should check existence in actual state to process here.

Reply:
If the volume was processed successfully in the previous iteration, it should be found via the local getUniqueVolumeName.

Reply:
i think it might be found or not from local getUniqueVolumeName depending on whether actual state has it or not. If for some reason, mount failed, actual state will not have this volume information.